web-marketplace/src/components/organisms/PostForm/PostForm.tsx

```diff
@@ -0,0 +1,212 @@
+import { Body, Title } from 'web-components/lib/components/typography';
+
+import { Link } from 'components/atoms';
+import Form from 'components/molecules/Form/Form';
+import { useEffect } from 'react';
+import { useFieldArray, useWatch } from 'react-hook-form';
+import ContainedButton from 'web-components/lib/components/buttons/ContainedButton';
+import { Radio } from 'web-components/lib/components/inputs/new/Radio/Radio';
+import { RadioGroup } from 'web-components/lib/components/inputs/new/RadioGroup/RadioGroup';
+import { TextAreaFieldChartCounter } from 'web-components/lib/components/inputs/new/TextAreaField/TextAreaField.ChartCounter';
+import { cn } from 'web-components/lib/utils/styles/cn';
+import OutlinedButton from 'web-components/src/components/buttons/OutlinedButton';
+import {
+  DEFAULT_URL,
+  IMAGE_UPLOAD_BUTTON_LABEL,
+  cropAspectMediaForm,
+} from '../MediaForm/MediaForm.constants';
+import { useMediaFormStyles } from '../MediaForm/useMediaFormStyles';
+import {
+  POST_MAX_TITLE_LENGTH,
+  POST_PRIVACY_DESCRIPTION,
+} from './PostForm.constants';
+import { PostFormSchemaType, postFormSchema } from './PostForm.schema';
+
+export interface Props {
+  initialValues: PostFormSchemaType;
+  className?: string;
+}
+
+export const PostForm = ({ initialValues, className }: Props): JSX.Element => {
+  const form = useZodForm({
+    schema: postFormSchema,
+    defaultValues: {
+      ...initialValues,
+    },
+    mode: 'onBlur',
+  });
+  const { classes } = useMediaFormStyles();
+  const { errors } = form.formState;
+  const { setValue } = form;
+
+  const imageDropCommonProps: Partial<ImageDropProps> = {
+    classes: { main: classes.fullSizeMedia },
+    buttonText: IMAGE_UPLOAD_BUTTON_LABEL,
+    fixedCrop: cropAspectMediaForm,
+  };
+
+  const title = useWatch({ control: form.control, name: 'title' });
+  const files = useWatch({ control: form.control, name: 'files' });
+  const privacy = useWatch({ control: form.control, name: 'privacyType' });
+
+  const noFiles = (files?.length ?? 0) <= 1;
+
+  const { fields, append, remove } = useFieldArray({
+    name: 'files',
+    control: form.control,
+  });
+
+  const setFiles = (value: string, fieldIndex: number): void => {
+    if (files?.[fieldIndex]?.['schema:url'] === DEFAULT_URL) {
+      append({
+        'schema:url': DEFAULT_URL,
+      });
+    }
+    setValue(`files.${fieldIndex}.schema:url`, encodeURI(value));
+  };
+
+  const getHandleDeleteWithIndex = async (fieldIndex: number) => {
+    remove(fieldIndex);
+  };
+
+  /* Effect */
+
+  useEffect(() => {
+    if (
+      fields?.length === 0 ||
+      fields?.every(field => field['schema:url'] !== DEFAULT_URL)
+    ) {
+      append({
+        'schema:url': DEFAULT_URL,
+      });
+    }
+  }, [append, fields]);
+
+  return (
+    <Form className={cn('max-w-[560px]', className)} form={form}>
+      <Title variant="h3" sx={{ textAlign: 'center' }} className="mb-50">
+        New post
+      </Title>
+      <TextField
```

Comment:
this is not using the same font-size (16px) as in the comment field:

![image](https://github.com/regen-network/regen-web/assets/3149358/9782159c-27f2-449b-88d9-e75b63485388)

Reply:
@erikalogie just realizing that this is just how things are at the moment (text in textarea input are smaller than in regular text input, see on the edit profile form for instance), is that intended or should we update all inputs to be use the same font sizes? it would make sense to have something consistent across the whole app.